### PR TITLE
Custom Function Error Output Cases

### PR DESCRIPTION
**Pull request by jiju-MS**
Custom Function Error Output Cases
-Jinlong

### CODE REVIEW

File: addins/CFSample/customfunctions.js [hunk 1]
```diff
@@ -303,6 +404,8 @@ CustomFunctions.associate('STOCK_PRICE_HISTORY', stockPriceHistory);
 CustomFunctions.associate('DISPLAY_WEB_DIALOG', displayWebDialog);
 CustomFunctions.associate('BITCOIN_STREAM', bitcoinStream);
 CustomFunctions.associate('ERROROUT', errorOut);
+CustomFunctions.associate('CUSTOM_ERROR_RETURN customErrorReturn', customErrorReturn);
```
Comment (zlatko-michailov) [inline addins/CFSample/customfunctions.js]: Pick one spelling. It looks like 'CUSTOM_ERROR_RETURN' is in line with the existing function IDs.

File: addins/CFSample/customfunctions.json [hunk 2]
```diff
@@ -342,6 +342,35 @@
       "options": {
         "stream": true
       }
+    },
+    {
+      "id": "customErrorReturn",
```
Comment (zlatko-michailov) [inline addins/CFSample/customfunctions.json]: The 'id' should match the first argument passed to `associate()`.

File: addins/CFSample/customfunctions.json [hunk 3]
```diff
@@ -342,6 +342,35 @@
...
+      "parameters": [
+        {
+          "name": "errorCase",
+          "description": "error case",
+          "type": "number"
+        }
+      ]
+    },
+    {
+      "id": "customErrorReturnArray",
```
Comment (zlatko-michailov) [inline addins/CFSample/customfunctions.json]: Before you send out the next iteration, please do the following:
1. Test your manifest from a file share. (You'll need to make the JSON metadata available from some web site.) Make sure the new functions get registered, and you can execute them.
2. This add-in is 'non-shared runtime'. We try to keep parity between add-ins - same functionality in a shared runtime and in a non-shared runtime. Consult with the owners - @madhavagrawal17 and @ylu0826 which one is the corresponding add-in.

Comment (jiju-MS) [reply addins/CFSample/customfunctions.json]: got it, will ask @madhavagrawal17 and @ylu0826 for help

File: addins/CFSample/customfunctions.js (hunk 1 above)
Comment (jiju-MS) [reply addins/CFSample/customfunctions.js]: spelling mistake. Will change to 'CUSTOM_ERROR_RETURN

File: addins/CFSample/customfunctions.json (hunk 3 above)
Comment (madhavagrawal17) [reply addins/CFSample/customfunctions.json]: @jiju-MS, Currently CFsample and CFsample2 (Shared Runtime) add-ins are not the same Custom functions so it is fine to not add this new API for CFsample2.

Comment (jiju-MS) [reply addins/CFSample/customfunctions.json]: Understand, so if we want to test Shared Runtime scenario, do we also need to update CFSample2?

File: addins/CFSample/customfunctions.json (hunk 2 above)
Comment (jiju-MS) [reply addins/CFSample/customfunctions.json]: Changed in the latest iteration.

File: addins/CFSample/customfunctions.js (hunk 1 above)
Comment (zlatko-michailov) [reply addins/CFSample/customfunctions.js]: You still have 2 words in the ID.

File: addins/CFSample/customfunctions.json (hunk 3 above)
Comment (zlatko-michailov) [reply addins/CFSample/customfunctions.json]: Yes. Perhaps not necessarily CFSample2, but one of the other CFSample add-ins that uses a Shared Runtime.

**Pull request merged**